### ISSUE TITLE
[data grid] Webworker export demo breaks

### ISSUE THREAD

**Issue by michelengelen** [mui/mui-x#13039]
### Search keywords

print export docs break

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: [export webworker demo](https://v6.mui.com/x/react-data-grid/export/#using-a-web-worker)

Steps:
1. try to export to print
2. demo hangs and eventually breaks

### Current behavior

Demo breaks

### Expected behavior

Demo should not break

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

used browser: Chrome

```
  System:
    OS: macOS 14.4.1
  Binaries:
    Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
    npm: 9.6.7 - ~/.nvm/versions/node/v18.17.1/bin/npm
    pnpm: 8.14.1 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 124.0.6367.119
    Edge: Not Found
    Safari: 17.4.1
```
</details>

**Comment by michelengelen** [comment]
Maybe we should disable the print export? 🤷🏼

**Comment by cherniavskii** [comment]
Yeah, print export doesn't support large datasets like this. This is the limitation of the current Print export implementation.
